### DISCUSSION TITLE
AKS Seq pods eviction

### DISCUSSION THREAD

**Discussion by venick** [datalust/seq-tickets#1585] [Help]
I'm looking for help or guidance with running Seq on kubernetes (specifically Azure AKS in our case).

About once a week the pod running Seq is being evicted with the message "**Pod The node had condition: [MemoryPressure]**". This often happens about 10 times in a row before becoming stable again for anther 5-10 days before it repeats again.

Seq version: 2022.1 (datalust/seq:2022.1.7449)
Node size: 2 vCPUs 8GB (B2ms)
Ingesting roughly 2.6 million events per day
Retention period: 9 days

I assume that increasing the node's memory or reducing events logged would help. However the first option has financial constraints and the second would reduce usefulness of the logs.

Are there any configurations, updates or other considerations I can address first, before needing to invest money in higher spec hardware?

**Comment by andymac4182** [comment]
You could try setting the `SEQ_CACHE_SYSTEMRAMTARGET` to reduce the amount of ram the that container uses. https://docs.datalust.co/docs/json-config#cachesystemramtarget

Also are the memory limits being set on the container level not just the pod level?

**Reply by venick** [reply]
Thanks @andymac4182 

SEQ_CACHE_SYSTEMRAMTARGET was set to '0.6' which would mean 60%. I'll try lowering it to 0.5 and see if that helps.
How does this configuration setting work? Is it a hard limit or can it peak over this for long enough that AKS decides to act?

> "Also are the memory limits being set on the container level not just the pod level?"

There were no resource limits specifically assigned but I've now added a request of 4Gi of memory to see if that makes a difference. Considering that Seq was running alone on that node, I'm not very confident it will make a great impact.

**Reply by andymac4182** [reply]
I have had that issue before. Seq needs the cgroup settings to be there to manage the memory properly. I have this issue on AWS Fargate as well. Setting memory limits on the container and pod solved the issues for me and I could let the SYSTEMRAMTARGET go back to defaults after that.
